datadog/resource_datadog_integration_pagerduty_service_object.go

```diff
@@ -0,0 +1,112 @@
+package datadog
+
+import (
+	"fmt"
+	"sync"
+
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/zorkian/go-datadog-api"
+)
+
+const maskedSecret = "*****"
+
+// creating/modifying/deleting multiple service objects in parallel on one account is unsupported
+// by the API right now; therefore we use the mutex to only operate on one at a time
+var integrationPdSOMutex = sync.Mutex{}
+
+func resourceDatadogIntegrationPagerdutySO() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceDatadogIntegrationPagerdutySOCreate,
+		Read:   resourceDatadogIntegrationPagerdutySORead,
+		Update: resourceDatadogIntegrationPagerdutySOUpdate,
+		Delete: resourceDatadogIntegrationPagerdutySODelete,
+		// since the API never returns service_key, it's impossible to meaningfully import resources
+		Importer: nil,
+
+		Schema: map[string]*schema.Schema{
+			"service_name": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"service_key": {
+				Type:      schema.TypeString,
+				Required:  true,
+				Sensitive: true,
+			},
+		},
+	}
+}
+
+func buildIntegrationPagerdutySO(d *schema.ResourceData) *datadog.ServicePDRequest {
+	so := &datadog.ServicePDRequest{}
+	if v, ok := d.GetOk("service_name"); ok {
+		so.SetServiceName(v.(string))
+	}
+	if v, ok := d.GetOk("service_key"); ok {
+		so.SetServiceKey(v.(string))
+	}
+
+	return so
+}
+
+func resourceDatadogIntegrationPagerdutySOCreate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*datadog.Client)
+	integrationPdSOMutex.Lock()
+	defer integrationPdSOMutex.Unlock()
+
+	so := buildIntegrationPagerdutySO(d)
+
+	if err := client.CreateIntegrationPDService(so); err != nil {
+		// TODO: warn user that PD integration must be enabled to be able to create service objects
+		return fmt.Errorf("Failed to create integration pagerduty using Datadog API: %s", err.Error())
```

Comment:
Should we link to the documentation here?

Reply:
And say that you need to install the tile first? (And/or show the TF page about enabling the integration globally?)

Reply:
The error returned from API says `Pagerduty Integration not found` (which is printed with the error message) is IMO sufficient. I think this is self-explanatory enough. Do you need it needs to be improved? Because if we wanted to provide the docs just in this case, we'd need to scan precisely for the `Pagerduty Integration not found` string and that can be error-prone if that string changes a bit in the future...

Reply:
Yeah thats totally valid. Lets keep as is then 🙂